hrm-domain/packages/hrm-search-config/generateElasticsearchQuery.ts

```diff
@@ -274,6 +274,69 @@ const generateTranscriptQueriesFromFilters = ({
   }));
 };
 
+const generateContactNumberQueries = ({
+  searchParameters,
+  buildParams = { parentPath: '' },
+  queryWrapper = p => p,
+}: {
+  searchParameters: SearchParametersContact;
+  buildParams?: { parentPath: string };
+  queryWrapper?: (
+    p: DocumentTypeQueryParams[DocumentType],
+  ) => DocumentTypeQueryParams[DocumentType];
+}): QueryDslQueryContainer[] => {
+  const defaultQueries = [
+    generateQueryFromSearchTerms({
+      documentType: DocumentType.Contact,
+      fields: [
+        {
+          field: 'number',
+          boost: BOOST_FACTORS.contact,
+        },
+      ],
+      searchTerm: searchParameters.searchTerm,
+      parentPath: buildParams.parentPath,
+      queryWrapper,
+    }),
+  ];
+  const numberTerm = searchParameters.searchTerm.match(/\d+/g)?.join('');
```

Comment:
Would this not extract all the numbers from the search term and put them together? Would it not be better to try to capture different possible numbers (i.e. groups of digits & phone number formatting characters over a certain length with no other characters) and sanitize & add each one to the search separately? Then we can support if they search multiple numbers, and there is less chance of mismatches

For example, in this logic, could somebody not search for `11 High Street, +44 7891 111 456` and then we would end up looking for +11447891111456 as a sanitised phone number match, when we need +447891111456?

Reply:
The problem I see with this idea is that it would be impossible to determine where to start and end different pieces of the string to compose the phone number, e.g.:
```
'11 High Street, +44 7891 111 456'.match(/\d+/g);
> ['11', '44', '7891', '111', '456']
```

If we use "contiguous strings" then `'11 High Street, +44-7891-111-456'` will behave as you describe, but the original example `11 High Street, +44 7891 111 456` won't.

I'm not against the idea of doing something smarter, I'm just having a hard time trying to decide which is the correct approach vs just assuming the user uses a single term for the search when using phone numbers.

Reply:
It doesn't have to get it right every time. If it finds and 'sanitises' something that isn't really a phone number, it's not the end of the world, and if it captures too much and sanitises something that doesn't match, it'll be a rare enough case that that's ok too, it doesn't have to be perfect

Reply:
Something as simple as running `[\d\+\s\-]{10,}` on the whole input and sanitising all the matches will do. Sure it will fail if there are space separated lists of numbers, or phone numbers and other numbers only separated by spaces, and it will sometimes capture and 'sanitise' things that aren't numbers, but those would be rare cases. It would catch common cases like a phone number before or after something else, or multiple phone numbers with a sensible separator like a comma. I think that's good enough

Reply:
@stephenhand I think I addressed this, thanks for the suggestion (and help)!